src/go/rpk/pkg/cli/cmd/cluster/config/export.go

```diff
@@ -0,0 +1,187 @@
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
+)
+
+func exportConfig(
+	file *os.File, schema admin.ConfigSchema, config admin.Config, all bool,
+) (err error) {
+	template := make(map[string]interface{})
+
+	// Present properties in alphabetical order, providing some pseudo-grouping based on common prefixes
+	keys := make([]string, 0, len(schema))
+	for k := range schema {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, name := range keys {
+		meta := schema[name]
+		curValue := config[name]
+		template[name] = curValue
+
+		visibility := meta.Visibility
+
+		if visibility == "deprecated" {
+			continue
+		}
+
+		if meta.Visibility == "tunable" && !all {
+			continue
+		}
+
+		// Preface each property with a descriptive comment
+		var comment_tokens []string = make([]string, 0)
+
+		if meta.Example != "" {
+			comment_tokens = append(comment_tokens, fmt.Sprintf("e.g. '%s'", meta.Example))
+		}
+
+		if meta.NeedsRestart {
+			comment_tokens = append(comment_tokens, "restart required")
+		}
+
+		if meta.Nullable {
+			comment_tokens = append(comment_tokens, "may be nil")
+		}
+
+		comment_details := ""
+		if len(comment_tokens) > 0 {
+			comment_details = fmt.Sprintf(" (%s)", strings.Join(comment_tokens, ", "))
+		}
+		file.Write([]byte(fmt.Sprintf("\n# %s%s\n", meta.Description, comment_details)))
+
+		// Compose a YAML representation of the property: this is
+		// done with simple prints rather than the yaml module, because
+		// in either case we have to carefully format values.
+		if meta.Type == "array" {
+			switch x := curValue.(type) {
+			case nil:
+				file.Write([]byte(fmt.Sprintf("%s: []", name)))
+			case []interface{}:
+				if len(x) > 0 {
+					file.Write([]byte(fmt.Sprintf("%s:\n", name)))
+					for _, v := range x {
+						file.Write([]byte(fmt.Sprintf("    - %v\n", v)))
+					}
+				} else {
+					file.Write([]byte(fmt.Sprintf("%s: []", name)))
+				}
+			default:
+				out.Die("Unexpected property value type: %s: %T", name, curValue)
+			}
+
+		} else {
+			scalarVal := ""
+			switch x := curValue.(type) {
+			case int:
+				scalarVal = strconv.Itoa(x)
+			case float64:
+				// Only format floats with a decimal point if they
+				// don't have an integer value currently
+				if x == float64(int64(x)) {
+					scalarVal = strconv.FormatInt(int64(x), 10)
+				} else {
+					scalarVal = strconv.FormatFloat(x, 'f', -1, 64)
```

Comment:
`'f'` doesn't look to print a decimal point if there is no fractional component, so I don't think this branch is necessary: https://go.dev/play/p/meAjeHI4zzL

Reply:
Interesting, the `yaml` crate mangled floats but the stock golang print doesn't.  removed this branch.